tests/contrib/kafka/test_kafka.py

```diff
@@ -0,0 +1,228 @@
+import confluent_kafka
+import six
+
+from ddtrace import Pin
+from ddtrace.constants import ANALYTICS_SAMPLE_RATE_KEY
+from ddtrace.contrib.kafka.patch import patch
+from ddtrace.contrib.kafka.patch import unpatch
+from tests.contrib.config import KAFKA_CONFIG
+from tests.utils import TracerTestCase
+from tests.utils import override_config
+
+
+class TestKafkaPatch(TracerTestCase):
```

Comment:
nit: I believe we're moving away from using the `TracerTestCase` class in our test suite. Can we add snapshot tests and use plain pytest functions.

Reply:
I'd love to convert this to snapshots, but unsure how to do so since there's a non-deterministic number of spans generated when the consumer waits for a message. Do you know how to account for non-deterministicness in a snapshot test?